python/tvm/relay/op/_tensor.py

```diff
@@ -80,3 +81,28 @@ def clip_compute(attrs, inputs, output_type, target):
     return [topi.clip(inputs[0], attrs.a_min, attrs.a_max)]
 
 register_schedule("clip", schedule_elemwise)
+
+# TODO: Move to C++.
+
+def add_grad(*args):
+   from tvm.relay import TypeVar, Function, Var, FuncType, TupleType, Tuple, const
+   # (Lhs, Rhs) -> Out
+   #   where Broadcast(Lhs, Rhs)
+   # (Lhs, Rhs) -> (Out, (Out) -> (Lhs, Rhs))
+   rhs_type = TypeVar('rhs')
+   lhs_type = TypeVar('lhs')
+   out_type = TypeVar('out')
+   rhs = Var('rhs', rhs_type)
+   lhs = Var('lhs', lhs_type)
+   ret_type = FuncType([out_type], TupleType([lhs_type, rhs_type]))
+   ret_type = TupleType([out_type, ret_type])
+   ret_type = FuncType([lhs_type, rhs_type], ret_type)
+   body = Tuple([const(1.0), const(1.0)])
```

Comment:
According to the comment above, we expect a Function returning a tuple of Value and another function, returning a tuple of gradients. It looks like `ret_type` definition does match this comment. But the `body` of function below is defined as a tuple of two constants. How is it possible?

Reply:
This isn't fully done, we are blocked on some issues with type checking.